src/main/java/org/beanpod/switchboard/controller/DecoderController.java

```diff
@@ -29,24 +31,35 @@
 @RequestMapping("/decoder")
 @RequiredArgsConstructor
 public class DecoderController {
+
   public static final String UNKNOWN_ERROR_MESSAGE = "Unknown error in DecoderController";
 
   private final DecoderDaoImpl decoderDao;
   private final DeviceDaoImpl deviceService;
   private final DecoderMapper decoderMapper;
   private final StreamMapper streamMapper;
   private final DecoderService decoderService;
+  private final MaintainDeviceStatus maintainDeviceStatus;
 
   @GetMapping
   public List<DecoderDto> retrieveAllDecoders() {
     List<DecoderEntity> decoderEntity = decoderDao.getDecoders();
+    maintainDeviceStatus.maintainStatusField(decoderEntity);
     return decoderMapper.toDecoderDtos(decoderEntity);
   }
 
   @GetMapping("/{serialNumber}")
   public ResponseEntity<DecoderDto> retrieveDecoder(@PathVariable @Valid String serialNumber) {
-    return decoderDao
-        .findDecoder(serialNumber)
+    // maintain status field and create a log if status changed
+    Optional<DecoderDto> decoder = decoderDao.findDecoder(serialNumber);
+    if (decoder.isPresent()) {
+      List<DecoderEntity> decodersListTemp = new LinkedList<>();
+      decodersListTemp.add(decoderMapper.toDecoderEntity(decoder.get()));
+      maintainDeviceStatus.maintainStatusField(decodersListTemp);
+    }
+
+    // return decoder
+    return decoder
```

Comment:
here i think you are returning the decoder that is not maintained, because you find the decoder first then you maintain the status. Id do as the other comment
- maintain the status 
- return the properly maintained status
- use the aspect to do the logging part "afterReturning"

Reply:
Good catch! I've updated the retrieveEncoder/retrieveDecoder controllers to return the most updated object.